dash.el

```diff
@@ -2274,9 +2274,11 @@ The test for equality is done with `equal',
 or with `-compare-fn' if that's non-nil.
 
 Alias: `-uniq'"
-  (let (result)
-    (--each list (unless (-contains? result it) (!cons it result)))
-    (nreverse result)))
+  (if (memq -compare-fn '(eq eql equal))
+      (-union list nil)
```

Comment:
Have you tested this? Here's what I currently get:

```el
(-uniq '(t t t)) ;; => (t)

(let ((-compare-fn #'eq))
  (-union '(t t t) nil))
  ;; => (t t t)
```

Reply:
Can we add some tests where we modify the `-compare-fn` to those tested in the predicate?  I'm getting the same results as @basil-conto so it would be nice to cover it with automated tests.